src/GeneralStateTestsFiller/stEWASMTests/utils/callCodeFiller.yml

```diff
@@ -0,0 +1,92 @@
+callCode:
+  env:
+    currentCoinbase: 2adc25665018aa1fe0e6bc666dac8fc2697ff9ba
+    currentDifficulty: '0x020000'
+    currentGasLimit: '89128960'
+    currentNumber: '1'
+    currentTimestamp: '1000'
+    previousHash: 5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6
+  pre:
+    # tx sender
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '100000000000'
+      code: ''
+      nonce: ''
+      storage: {}
+    # called contract
+    deadbeef00000000000000000000000000000000:
+      balance: '100000000000'
+      code: |
+        (module
+          (import "ethereum" "storageStore" (func $storageStore (param i32 i32)))
+          (memory 1)
+          (export "memory" (memory 0))
+          (export "main" (func $main))
+          (func $main
+            ;; Path
+            (i32.store (i32.const 0) (i32.const 0))
+
+            ;; Value
+            (i32.store (i32.const 32) (i32.const 0xabcd))
+
+            ;; Write to storage
+            (call $storageStore (i32.const 0) (i32.const 32))
+          )
+        )
+      nonce: ''
+      storage: {}
+    # main contract, tx receiver
+    abcdabcdabcdabcdabcdabcdabcdabcdabcdabcd:
+      balance: '100000000000'
+      code: |
+        (module
+          (import "ethereum" "getExternalCodeSize" (func $getExternalCodeSize (param i32) (result i32)))
+          (import "ethereum" "getAddress" (func $getAddress (param i32)))
+          (import "ethereum" "storageStore" (func $storageStore (param i32 i32)))
+          (import "ethereum" "callCode" (func $callCode (param i64 i32 i32 i32 i32 i32 i32) (result i32)))
+
+          (memory 1)
+          (data (i32.const 0)  "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\ef\be\ad\de")
+
+          (export "memory" (memory 0))
+          (export "main" (func $main))
+
+          (func $main
+            (i32.store (i32.const 96) (call $callCode (i64.const 100000) (i32.const 0) (i32.const 32) (i32.const 32) (i32.const 1) (i32.const 64) (i32.const 32)))
+
+            ;; Expect that the call did not trap i.e. result was 1
```

Comment:
Trapping here does nothing in terms of affecting the state and changing the test outcome.  Maybe add a store statement after this and add an expect case for `abcdabcdabcdabcdabcdabcdabcdabcdabcdabcd` to have storage after trap statement be set.

Reply:
"trap" is the wrong word here. Basically I think we should check that `callCode` had a result of 1 to indicate success. I think your comment here refers to the conditional that follows and the `(unreachable)`. I got this syntax from the `ewasm-kernel` tests e.g. https://github.com/ewasm/ewasm-kernel/blob/master/tests/interface/call.wast. Is there a reason those tests include this syntax but it's not useful here?